src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,65 @@
-import React from 'react';
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import { HandleCommentAdd } from '../types/handlers';
+
+type Props = {
+  onCommentAdd: HandleCommentAdd;
+};
+
+export const NewCommentForm: React.FC<Props> = ({ onCommentAdd }) => {
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [text, setText] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [sendError, setSendError] = useState(false);
+  const [nameError, setNameError] = useState(false);
+  const [emailError, setEmailError] = useState(false);
+  const [textError, setTextError] = useState(false);
+
+  const handleNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setNameError(false);
+    setName(event.currentTarget.value);
+  };
+
+  const handleEmailChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setEmailError(false);
+    setEmail(event.currentTarget.value);
+  };
+
+  const handleTextChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setTextError(false);
+    setText(event.currentTarget.value);
+  };
+
+  const handleClearClick = () => {
+    setNameError(false);
+    setName('');
+    setEmailError(false);
+    setEmail('');
+    setTextError(false);
+    setText('');
+  };
+
+  const handleSubmit = async (event: React.FormEvent) => {
+    event.preventDefault();
+    setIsLoading(true);
+    setSendError(false);
+
+    const errors = await onCommentAdd(name, email, text);
+
+    setSendError(errors.sendError);
+    setNameError(errors.nameError);
+    setEmailError(errors.emailError);
+    setTextError(errors.textError);
+    setIsLoading(false);
+
+    if (Object.values(errors).every(formError => !formError)) {
+      setText('');
+    }
+  };
 
-export const NewCommentForm: React.FC = () => {
   return (
-    <form data-cy="NewCommentForm">
+    <form data-cy="NewCommentForm" onSubmit={handleSubmit}>
       <div className="field" data-cy="NameField">
         <label className="label" htmlFor="comment-author-name">
           Author Name
```

Comment:
Breaking the input fields into separate components can improve code readability and reusability. Let's create a reusable InputField and TextareaField component for the form inputs.